jOOL-java-8/src/test/java/org/jooq/lambda/CollectorTests.java

```diff
@@ -308,7 +297,225 @@ public void testPercentileWithStringsAndFunction() {
+            final int val;
+            Item(int val) {
+                this.val = val;
+            }
+        }
+
+        Item    a = new Item(1), b = new Item(1), c = new Item(2), d = new Item(2),
+                e = new Item(2), f = new Item(3), g = new Item(3), h = new Item(3),
+                i = new Item(4), j = new Item(4), k = new Item(5), l = new Item(6),
+                m = new Item(7), n = new Item(7), o = new Item(7), p = new Item(7);
+
+        assertEquals(asList(h, g, f), Seq.of(c, j, n, d, e, o, l, p, a, m, h, b, k, g, f, i).collect(medianAllBy(item -> item.val)).toList());
+    }
+
+    //CS304 (manually written) Issue link: https://github.com/jOOQ/jOOL/issues/221
+    @Test
+    public void testMedianAllWithComparator() {
+        assertEquals(asList(2, 2, 2, 2), Seq.of(1, 3, 2, 1, 2, 1, 2, 3, 3, 2).collect(medianAll(Comparator.comparingInt(o -> -o))).toList());
+    }
+
+    //CS304 (manually written) Issue link: https://github.com/jOOQ/jOOL/issues/221
+    @Test
+    public void testMedianAllWithComparator2() {
+        class Item implements Comparable<Item>{
+            final int val;
+            Item(int val) {
+                this.val = val;
+            }
+
+            @Override
+            public int compareTo(Item o) {
+                return this.val - o.val;
+            }
+
+            Item reverse() {
+                return new Item(-this.val);
+            }
+        }
+
+        Item    a = new Item(1), b = new Item(1), c = new Item(2), d = new Item(2),
+                e = new Item(2), f = new Item(3), g = new Item(3), h = new Item(3),
+                i = new Item(4), j = new Item(4), k = new Item(5), l = new Item(6),
+                m = new Item(7), n = new Item(7), o = new Item(7), p = new Item(7);
+
+        assertEquals(asList(j, i), Seq.of(c, j, n, d, e, o, l, p, a, m, h, b, k, g, f, i).collect(medianAll(Comparator.comparing(Item::reverse))).toList());
+    }
+
+    //CS304 (manually written) Issue link: https://github.com/jOOQ/jOOL/issues/221
+    @Test
+    public void testMedianAllWithoutComparator() {
+        Supplier<Seq<Integer>> s = () -> Seq.of(1, 3, 2, 1, 2, 1, 2, 3, 3, 2);
+
+        assertEquals(asList(1, 1, 1), s.get().collect(percentileAll(0.25)).toList());
+        assertEquals(asList(2, 2, 2, 2), s.get().collect(percentileAll(0.5)).toList());
+        assertEquals(asList(3, 3, 3), s.get().collect(percentileAll(0.75)).toList());
+    }
+
+    //CS304 (manually written) Issue link: https://github.com/jOOQ/jOOL/issues/221
+    @Test
+    public void testMedianAllWithoutComparator2() {
+        class Item implements Comparable<Item>{
+            final int val;
+            Item(int val) {
+                this.val = val;
+            }
+
+            @Override
+            public int compareTo(Item o) {
+                return this.val - o.val;
+            }
+
+            Item reverse() {
+                return new Item(-this.val);
+            }
+        }
```

Comment:
I'll merge this, but I'll probably get rid of this class. I'm not sure what value it provides compared to just using `Integer`

Reply:
Nevermind. I see the point now. I'll just move the local class out of the method for better reuse.